delivery-service/ds/src/main.rs

```diff
@@ -173,10 +178,6 @@ async fn publish_key_packages(
         None => return actix_web::HttpResponse::NotFound().finish(),
     };
 
-    let mut bytes = web::BytesMut::new();
-    while let Some(item) = body.next().await {
-        bytes.extend_from_slice(&unwrap_item!(item));
-    }
```

Comment:
Is there a particular reason you are moving this up to the beginning of the function?

Reply:
Yes, there was a clippy warning that we were holding a MutexGuard across an .await